### PR TITLE
phpstan lvl 1

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

**Comment by trasher** [comment]
No idea why `cd` is not found, I cannot make phpstan checks work.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -95,3 +100,5 @@ jobs:
           docker exec ${{ job.services.app.id }} bin/console glpi:database:install --ansi --no-interaction --db-name=glpi --db-host=db --db-user=root --strict-configuration
           docker exec ${{ job.services.app.id }} bin/console glpi:plugin:install --ansi --no-interaction --username=glpi oauthimap
           docker exec ${{ job.services.app.id }} bin/console glpi:plugin:activate --ansi --no-interaction oauthimap
+      - name: PHPStan checks
+        run: docker exec ${{ job.services.app.id }} cd plugins/oauthimap && php vendor/bin/phpstan analyze --ansi --memory-limit=1G --no-interaction --no-progress
```
Comment (cedric-anne) [inline .github/workflows/continuous-integration.yml:104]:
```suggestion
      - name: "PHPStan checks"
        run: docker exec ${{ job.services.app.id }} bash -c "cd plugins/oauthimap && php vendor/bin/phpstan analyze --ansi --memory-limit=1G --no-interaction --no-progress"
```

File: inc/mailcollectorfeature.class.php [hunk 2]
```diff
@@ -308,10 +308,11 @@ public static function updateMailCollectorOnAuthorizationCallback(
                         'login' => $authorization->fields['email'],
                     ]
                 );
+                Html::redirect($mailcollector->getLinkURL());
```
Comment (cedric-anne) [inline inc/mailcollectorfeature.class.php:311]: We should try to redirect to `$mailcollector->getLinkURL()` even if `$success === false`, and if we do not have a valid mailcollector ID (this is not an expected usecase), we should redirect to the mail collector list. Indeed `Html::back()` will probably redirect user on the last page of the external provider OAuth authorization page, and is may result in a redirection loop.

```php
        $mailcollector = new MailCollector();
        $mailcollector_id = $params[$mailcollector->getForeignKeyField()] ?? null;
        if ($mailcollector_id !== null && $mailcollector->getFromDB($mailcollector_id)) {
            if ($success) {
                // Store authorized email into MailCollector
                $mailcollector->update(
                    [
                        'id'    => $mailcollector_id,
                        'login' => $authorization->fields['email'],
                    ]
                );
            }
            Html::redirect($mailcollector->getLinkURL());
        }

        Html::back($mailcollector->getSearchURL());
```

Comment (trasher) [reply inc/mailcollectorfeature.class.php]: I'm pretty sure getLinkURL will not work as expected if getFromDB fails; see last commit

**Comment by trasher** [comment]
Currently failing with `Stub file /var/glpi/stubs/glpi_constants.php does not exist.`; but it should exists, path seems corrrect

**Comment by cedric-anne** [comment]
> Currently failing with `Stub file /var/glpi/stubs/glpi_constants.php does not exist.`; but it should exists, path seems corrrect

The current CI process uses the nightly build of GLPI, and this nightly build does not include the "development"  files, like stubs, tests, ...

I will try to produce soon a "plugins CI" image that contains a "dev" version of GLPI. It would probably help to speed up CI on all plugins, and simplify the way GLPI is retrieved.

**Pull request closed without merging**